net/dns/resolver/tsdns.go

```diff
@@ -650,14 +652,28 @@ func (r *Resolver) parseViaDomain(domain dnsname.FQDN, typ dns.Type) (netip.Addr
 
 	var siteID string
 	var ip4Str string
-	if strings.HasPrefix(fqdn, "via-") {
+	switch {
+	case strings.Contains(fqdn, "-via-"):
+		// Format number 3: "192-168-1-2-via-7" or "192-168-1-2-via-7.foo.ts.net."
```

Comment:
If we do eventually implement the named site-ids proposal I think this would be `192-168-1-2-via-toronto.foo.ts.net`, which seems fine.